### PR TITLE
CPhaseLeakageAnalysis: do not subtract the reference data in leakage extraction

### PR DESCRIPTION
**Pull request by chellings**
A change by @stephlazar on XLD.

### CODE REVIEW

File: pycqed/analysis_v2/timedomain_analysis.py [hunk 1]
```diff
@@ -5255,7 +5255,7 @@ def analyze_fit_results(self):
         if self.leakage_qbname is not None:
             # get leakage
             if self.get_param_value('classified_ro', False):
-                leakage = self.leakage_values[0::2] - self.leakage_values[1::2]
+                leakage = self.leakage_values[0::2]
```
Comment (nathlacroix) [inline pycqed/analysis_v2/timedomain_analysis.py:5249]: @stephlazar  why is this change required? Isn't the goal to look at the leakage introduced by the gate, rather than the absolute leakage? The absolute leakage, we can get from the single qubit readout calibration (as we only do single qubit gates before the cphase gate)

**Review by nathlacroix — COMMENTED** [review]
@stephlazar would be great if you could explain a bit more about this pull request so that we can understand whether that's really something we want in master or if it's a debug/setup specific thing :)

**Comment by stephlazar** [comment]
We decided to remove the reference data, i.e. with the interaction off to reduce noise. Basically that data is just noise, and when we reduce leakage due to the gate, at some point we subtract two very small quantities that are mostly noise.

**Comment by chellings** [comment]
any conclusions here? @nathlacroix @antsr @christiankraglundandersen

**Comment by christiankraglundandersen** [comment]
I suggest to merge this PR @chellings

**Comment by nathlacroix** [comment]
@stephlazar I don't see how it reduce the noise to show "absolute" leakage. Typically the measured leakage is always on the order of one or several percent, due to readout overlap. Therefore, for a good gate the readout errors often dominates compared to leakage originating from the gate itself. The "relative" leakage leakage might in that case indeed be dominated by noise (but the noise level is independent of substracting the reference data), but it will still be close to 0 and I think this makes much more sense than showing a 3% leakage error which essentially comes from readout errors. This would lead people to think that they have 3% leakage on the gate, which is just wrong. 

We could of course do readout correction, but often, no readout correction is made, so we would essentially not distinguish the leakage from the gate and the leakage resulting from readout errors. Sure, we can mentally "substract" the leakage due to readout overlap but the user needs to know it mentally and I don't think it makes sense, as the goal is to characterize the leakage of the gate. 

In short, I am not in favor of merging this into the master, but think it should rather be kept on the S17 project branch. If I'm the only one with this opinion, then democracy wins and someone can then merge it into the master, but currently I don't see why this is needed and/or better than the current code in the master.

Other opinions, @chellings @antsr  ?

**Comment by chellings** [comment]
@nathlacroix 's argument sounds reasonable to me, but I do not really understand @stephlazar 's argument. So not sure whether I can judge this.

**Comment by christiankraglundandersen** [comment]
while @nathlacroix  comment makes sense, I am still with @stephlazar  on this one because this number is typically what we use as a measure when we optimize the gate. if both the leakage and the reference leakage is dominated by RO errors as in @nathlacroix 's example, then you subtract the two numbers and you are left with just noise. But this noise is now sqrt(2) times larger than if you did not subtract the two numbers, so your optimizer becomes less reliable. 

So I think it boils down to this: 
if you want to use the number to optimize your gate: go with @stephlazar 
if you want the number to be meaningful for the user: go with @nathlacroix

**Comment by nathlacroix** [comment]
Ok, thanks @christiankraglundandersen for the explanation, it does help clarify the usecase. While I'm not sure whether this is currently the dominating source of noise (compared to fluctuating perf, etc.), I do see the point. As a solution, I would suggest that @stephlazar adds a new label to the figure (i.e. for the user) that specifies the leakage with respect to reference (e.g. called "leakage w.r.t. ref: xxx%"), but that we use the "absolute" leakage for the optimization (which would keep the "leakage" label to avoid having to rename stuff).

**Comment by stephlazar** [comment]
I have now implemented @nathlacroix's suggestion to compute both quantities and store and show both on the plots. 

Leakage_qb4_pf_20200702_214915

image

**Review by stephlazar — APPROVED** [review]
The agreed-upon compromise has now been implemented and tested.
